.github/workflows/test.yml

```diff
@@ -8,6 +8,18 @@ on:
       - '**'
       - '!doc/**'
 jobs:
+  sanity:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-go@v2
+        with:
+          go-version: '~1.16'
+      - name: Install goimports
+        run: go install golang.org/x/tools/cmd/goimports@latest
+      - name: Run sanity checks
+        run: make vendor && make lint && git diff --exit-code
```

Comment:
I can also just split this out to run the linting check and remove that `lint` make target:

```yaml
goimports -w $(find . -name '*.go' -not -path "./vendor/"
```

And then have the `git diff ...` check in another step. Open to suggestions on how to best organize this, or whether we're fine with throwing this under a single step.

Reply:
Just my personal preference, but I like to have make targets that encapsulate the github jobs so that things are eay to reproduce locally with as little hassle as possible, including getting and using the correct version of whatever tool the Makefile depends on.

Not something I would hold this up for at all, but in SDK we ended up with [this script](https://github.com/operator-framework/operator-sdk/blob/master/tools/scripts/fetch) and then make targets would call it like this: https://github.com/operator-framework/operator-sdk/blob/4e16f401032ca61f1e57203840dbc303de365f42/Makefile#L126

It was nice because:
1. If we ever needed to switch to a different CI runner (which happened three times between Travis/Openshift CI and then to GitHub actions), it was pretty simple.
1. Anyone calling the make target is using the version specified, rather than whatever is on their path.
2. No one has to worry about figuring out how to get the various tools in the first place.

Whenever we had a new tool dependency, it was pretty easy to tack it onto the end of that case block in the fetch script.

Reply:
Yeah, absolutely that makes a lot of sense. I'll dive into those links some more - I think I would prefer getting something merged sooner-than-later just because this PR also changes files after running `goimports` locally, so it seems like some editors aren't fully set up to format on save. Going forward, and there's already a TODO in this PR about running `go vet ...` as well, we should probably aim towards emulating some of that SDK workflow.